offix/src/main/java/org/aerogear/offix/Offix.kt

```diff
@@ -182,6 +226,36 @@ fun getDao() = Offline.getDb()?.mutationDao()
  */
 fun inputTypeChecker(string: String) = string.equals("com.apollographql.apollo.api.Input")
 
+/*
+ This interceptor should be used by the user while making an instance of apollo client
+ which will help in detecting conflicts.
+ */
+fun getResponseInterceptor(): Interceptor? {
```

Comment:
Let's have a separate class called ConflictInterceptor.

The users can then add the interceptor in their apps like this : 

```
apolloClient.addInterceptor(new ConflictInterceptor())
```

The current method doesn't look very good.

Reply:
All the logic you have here should go inside the ConflictInterceptor class that extends an Interceptor